src/main/kotlin/org/rust/lang/core/dfa/ControlFlowGraph.kt

```diff
@@ -338,8 +338,13 @@ private class ExitPointVisitor(
 
     private val RsExpr.isInTailPosition: Boolean
         get() {
+            // Ignore the expression if any of its ancestors are cfg-disabled
+            if (ancestors.any { !existsAfterExpansion }) return false
```

Comment:
`psi.existsAfterExpansion` implies that all its ancestors are also `existsAfterExpansion`, so checking this for all ancestors is not needed (and also it's kind of O(n^2)).

Reply:
oh, right. thanks!